File: 7/29/3.cpp
#include <string>
#include <iostream>
using namespace std;

bool solution(string s)
{
    bool answer = true;


    int p = 0 ;
    int y = 0;

    for(int i = 0; i <s.length() ; i ++){
        if(s[i] == 89 || s[i] == 121){
            y++;
        }
        else if(s[i] == 80 || s[i] == 112){
            p++;
        }

    }

    answer = p==y;

    cout<<p<<y;




    return answer;
}

File: 2021/3/30/3.py

class Solution:
    def findReplaceString(self, S: str, indexes, sources, targets) -> str:



        answer = S
        arr = []

        plusIndex = 0
        for idx,source in enumerate(sources) :
            print(S[indexes[idx]:indexes[idx]+len(source)])
            if S[indexes[idx]:indexes[idx]+len(source)] == source :
                arr.append([indexes[idx],len(source),len(targets[idx]),targets[idx]])

        arr = sorted(arr)

        for a in arr :
            answer = answer[:a[0]+plusIndex] + a[3] + answer[a[0]+plusIndex+a[1]:]
            plusIndex += a[2]-a[1]
            print(answer,plusIndex)


        return answer